src/tools/rustfmt/src/visitor.rs

```diff
@@ -533,6 +535,28 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
                 ast::ItemKind::Static(..) | ast::ItemKind::Const(..) => {
                     self.visit_static(&StaticParts::from_item(item));
                 }
+                ast::ItemKind::ConstBlock(ast::ConstBlockItem {
+                    id: _,
+                    span,
+                    ref block,
+                }) => {
+                    let context = &self.get_context();
+                    let offset = self.block_indent;
+                    self.push_rewrite(
+                        item.span,
+                        block
+                            .rewrite(
+                                context,
+                                Shape::legacy(
+                                    context.budget(offset.block_indent),
+                                    offset.block_only(),
+                                ),
+                            )
+                            .map(|rhs| {
+                                recover_comment_removed(format!("const {rhs}"), span, context)
+                            }),
+                    );
```

Comment:
I think you're going to need to call `rewrite_block()` directly here instead of `block.rewrite()` so that you can pass along attributes from the Item. I wan to avoid inner attributes getting removed like they were in https://github.com/rust-lang/rustfmt/issues/6158.